lib/auth/init.go

```diff
@@ -484,6 +483,49 @@ func Init(cfg InitConfig, opts ...ServerOption) (*Server, error) {
 	return asrv, nil
 }
 
+func initSetAuthPreference(asrv *Server, newAuthPref services.AuthPreference) error {
+	storedAuthPref, err := asrv.GetAuthPreference()
+	if err != nil {
+		if !trace.IsNotFound(err) {
+			return trace.Wrap(err)
+		}
+		storedAuthPref = nil
```

Comment:
Wouldn't `storedAuthPref` be `nil` if above method returned an error so this assignment is redundant?